Feature: Adds a flexible data table to view DC details

martinctc opened the pull request:
# Summary
This branch adds a flexible data table to the 'List of DCs' tab, and makes related changes to the data cleaning script which the flex data table uses. 

# Changes
The changes made in this PR are:
1. Added a DT table to the 'List of DCs' tab
1. Added a 'Details' placeholder to the app side bar
1. As part of the data cleaning script, concatenate columns to enable better display on the DT table
1. Added a Chinese translation tweak to the main 'Overview' tab of the app (dropdown)
1. 'District' and 'Region' columns are now available to the Shiny app, via the read in Google Sheets
...

# Check
- [x] The CI/CD travis.ci and R CMD checks pass.
- [x] The app can be deployed.
- [x] The data table displays with no errors. 
...

martinctc committed:
feat: Add Details tab for data table

- Placeholder; names TBC

martinctc committed:
feat: Add DT table to tab_dclist

martinctc committed:
format: Add Chinese translation for dropdown

martinctc committed:
feat: Clean up data table

- Changes on the side of data prepping script
- Change on the DT function on server.R

martinctc committed:
fix: map to updated Google Sheet

- Changed data cleaning script to ingest new sheet

martinctc committed:
feat: add links to email and GitHub

- Logo files also added under www

avisionh commented on www/HKDC.svg:
```diff
@@ -0,0 +1,120 @@
+           width="985.71429"
+           height="422.85715"
+           x="45.714287"
+           y="75.500565"
+           style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:133.33332825px;line-height:2;font-family:'Segoe UI';-inkscape-font-specification:'Segoe UI';stroke-width:0.75131482" /></flowRegion><flowPara
+         id="flowPara821-9"
+         style="font-style:normal;font-variant:normal;font-weight:bold;font-stretch:normal;font-size:133.33332825px;line-height:2;font-family:'Segoe UI';-inkscape-font-specification:'Segoe UI Bold';stroke-width:0.75131482">COUNCIL</flowPara><flowPara
+         style="font-style:normal;font-variant:normal;font-weight:bold;font-stretch:normal;font-size:133.33332825px;line-height:2;font-family:'Segoe UI';-inkscape-font-specification:'Segoe UI Bold';stroke-width:0.75131482"
+         id="flowPara847" /></flowRoot>    <flowRoot
+       xml:space="preserve"
+       id="flowRoot815-2-0"
+       style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:133.33332825px;line-height:2;font-family:'Segoe UI';-inkscape-font-specification:'Segoe UI';letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:0.75131482"
+       transform="matrix(0.35216042,0,0,0.35216042,71.817955,201.53644)"><flowRegion
+         id="flowRegion817-6-5"
+         style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:133.33332825px;line-height:2;font-family:'Segoe UI';-inkscape-font-specification:'Segoe UI';stroke-width:0.75131482"><rect
+           id="rect819-4-6"
+           width="985.71429"
+           height="422.85715"
+           x="45.714287"
+           y="75.500565"
+           style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:133.33332825px;line-height:2;font-family:'Segoe UI';-inkscape-font-specification:'Segoe UI';stroke-width:0.75131482" /></flowRegion><flowPara
+         style="font-style:normal;font-variant:normal;font-weight:bold;font-stretch:normal;font-size:133.33332825px;line-height:2;font-family:'Segoe UI';-inkscape-font-specification:'Segoe UI Bold';stroke-width:0.75131482"
+         id="flowPara847-6">INFO</flowPara></flowRoot>    <flowRoot
+       xml:space="preserve"
+       id="flowRoot878"
+       style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:40px;line-height:1.25;font-family:'Cambria Math';-inkscape-font-specification:'Cambria Math';letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:0.6830135"
+       transform="matrix(0.38737645,0,0,0.38737645,-3.3575162,114.80844)"><flowRegion
+         id="flowRegion880"
+         style="stroke-width:0.6830135"><rect
+           id="rect882"
+           width="288.57144"
+           height="431.42856"
+           x="40"
+           y="86.929138"
+           style="stroke-width:0.6830135" /></flowRegion><flowPara
+         id="flowPara884"
+         style="font-style:normal;font-variant:normal;font-weight:bold;font-stretch:normal;font-size:160px;font-family:'Microsoft YaHei';-inkscape-font-specification:'Microsoft YaHei Bold';stroke-width:0.6830135">香</flowPara><flowPara
+         id="flowPara886"
+         style="font-style:normal;font-variant:normal;font-weight:bold;font-stretch:normal;font-size:160px;font-family:'Microsoft YaHei';-inkscape-font-specification:'Microsoft YaHei Bold';stroke-width:0.6830135">港</flowPara></flowRoot>  </g>
+</svg>
```
Do we need the `.svg` file? I can only see the `.png` file being used.

martinctc replied:
It was used to create the .png image - it could be useful if we have future applications / versions of the logo? Could remove it if we want to keep the repo 'minimal', no strong preference either way!

avisionh replied:
Let's keep it then!

avisionh committed:
refactor: Hide Details tab

This is because it is currently not being used. Hide so users have no expexpectation on what it is for.

avisionh committed:
docs: Update Guidance tab with relevant links

Also remove link to personal email as this is in top-right now.

avisionh commented on global.R:
```diff
@@ -32,4 +32,17 @@ data_master_raw <-
\ No newline at end of file
+  mutate(exists_fb = if_else(condition = !is.na(x = facebook), "blue", "black"))
+
+# Version to be shown with DT on app
+data_master_details <-
+  googlesheets4::read_sheet(ss = sheet_url,
+                            sheet = "Master",
+                            na = c("N/A", "")) %>%
+  select(Constituency = "DropDownText",
+         Party,
+         DC,
+         Region,
+         District,
+         FB = "facebook")
```
I'd suggest we remove this code as it's doing the same stuff as `data_master_raw`, just extracting specific columns and renaming the `DropDownText` and `facebook` ones.

A better performing approach would do the renaming in the `server.R` file where you call `data_master_details` (Line 89). Here, you can do something like:

> `renderDT(`
>       `select(.data = data_master_details,`
>                   `Constituency = DropDownText,`
>                   `Party,`
>                   `DC,`
>                   `Region,`
>                   `District,`
>                   `FB = Facebook)`

martinctc replied:
Makes total sense, and agreed - now implemented

avisionh committed:
style: Add bracket end highlighters

This is be explicit in what the end of brackets refer to.

avisionh committed:
style: Rename headers in code

This is to make them more explicit to what tabs they relate to in the web app.

avisionh committed:
chore: Remove shiny deployment file

This adds no value to repo.

avisionh commented on .development/googlesheets_dataclean.R:
```diff
@@ -4,16 +4,28 @@ library(janitor)
 # define google sheets ----------------------------------------------------
-list_arguments <- list(sheet = c("Central & Western", "Wan Chai", "Eastern", "Southern"))
+dc_sheet_names <- sheet_names(sheet_url)
+dc_sheet_names2 <- dc_sheet_names[!grepl(x = dc_sheet_names, pattern = "^Master$|^DistrictCouncilKey$")]
```
Could you add a comment to say what this regex is doing?

martinctc replied:
Good shout, I'll add that

avisionh committed:
refactor: Name function correctly

Previously mispelt dat when it should be data. Also add commenting to describe what purrr is doing to load in each sheet into a dataframe into a list.

avisionh commented on .development/googlesheets_dataclean.R:
```diff
@@ -4,36 +4,54 @@ library(janitor)
 library(googlesheets4)
 
-sheet_url <- "https://docs.google.com/spreadsheets/d/1usk9Q-5lA4bL_z6KXpUohc_2x_KhDgLxtm-YEtim_yk/"
+sheet_url <- "https://docs.google.com/spreadsheets/d/1007RLMHSukSJ5OfCcDJdnJW5QMZyS2P-81fe7utCZwk/"
```
There are several times we are explicitly calling the name of the GSheet. A better approach is to specify this in an `.Renviron` file and call it from there. This way, if the GSheet link changes, we won't need to change it in each of the scripts that call it, we just need to change it in the `.Renviron` file. 

Nothing to worry about now. Probably something to pick up in a different PR.

avisionh commented on global.R:
```diff
@@ -21,7 +21,7 @@ library(DT)
 
 ## Read data from Google Sheets 4
-sheet_url <- "https://docs.google.com/spreadsheets/d/1usk9Q-5lA4bL_z6KXpUohc_2x_KhDgLxtm-YEtim_yk/"
+sheet_url <- "https://docs.google.com/spreadsheets/d/1007RLMHSukSJ5OfCcDJdnJW5QMZyS2P-81fe7utCZwk/"
```
There are several times we are explicitly calling the name of the GSheet. A better approach is to specify this in an `.Renviron` file and call it from there. This way, if the GSheet link changes, we won't need to change it in each of the scripts that call it, we just need to change it in the `.Renviron` file. 

Nothing to worry about now. Probably something to pick up in a different PR.

avisionh reviewed (commented):
Thanks for this, some really good progress. It's looking very good, slick and professional now.

Just a few prompts before I give explicit approval. Particularly:

1. Removing the importing of the spreadsheet in `global.R`.
1. Do we need the `www/HKDC.svg` file? I don't see it being called in the script. Is there a need to have it in the repo?
1.  Since we have a `www/HKDC-white.png' file, should we rename `www/HKDC.png` to `www/HKDC-black.png` to make it more explicit?
1. Would be good if the `.development/googlesheets_dataclean.R` script had a brief description of what the regex is doing.

martinctc committed:
docs: add comments for regex line

martinctc committed:
chore: remove redundant GSheet reads

- Update server.R for the DT to read from data_master_raw, instead of data_master_details

martinctc committed:
Merge branch 'feature/dc-data-table' of https://github.com/avisionh/dashboard-hkdistrictcouncillors into feature/dc-data-table
